src/components/Proposal/ProposalData.tsx

```diff
@@ -122,27 +139,32 @@ export default withSubscription({
   createObservable: async (props) => {
     const arc = getArc();
     const { currentAccountAddress, daoState, proposalId } = props;
-    const arcDao = daoState.dao;
-    const proposal = arc.proposal(proposalId);
-    await proposal.fetchStaticState();
-    const spender = proposal.staticState.votingMachine;
+    const arcDao = new DAO(arc, daoState);
+    const proposal = await Proposal.create(arc, proposalId);
+    await proposal.fetchState();
+    const spender = proposal ? proposal.coreState.votingMachine : "0x0000000000000000000000000000000000000000";
```

Comment:
this is really an arc.js question:  `proposal.fetchState` hydrates `proposal.coreState`?  Why then isn't it called `proposal.fetchCoreState` or `proposal.hydrateCoreState`?

("fetch" usually implies returning what you are fetching, which isn't happening in this function, or is it?)

Reply:
Unsure, and am indifferent. @jellegerbrandy do you have an opinion?